plugins/module_utils/network/sonic/config/ars/ars.py

```diff
@@ -0,0 +1,617 @@
+                        profile_dict['load-scaling-factor'] = enum_dict[profile_dict['load-scaling-factor']]
+                    portprofile_list.append({'name': profile['name'], 'config': profile_dict})
+                if portprofile_list:
+                    ars_dict['ars-port-profile'] = {'portprofile': portprofile_list}
+
+            if cmds.get('port_bindings'):
+                portbind_list = []
+                for bind in cmds['port_bindings']:
+                    portbind_list.append({'name': bind['name'], 'config': bind})
+                if portbind_list:
+                    ars_dict['ars-port-bind'] = {'portbind': portbind_list}
+
+            if cmds.get('ars_objects'):
+                arsobject_list = []
+                for obj in cmds['ars_objects']:
+                    obj_dict = self.get_renamed_dict(obj)
+                    if obj_dict.get('mode'):
+                        obj_dict['mode'] = enum_dict[obj_dict['mode']]
+                    arsobject_list.append({'name': obj_dict['name'], 'config': obj_dict})
+                if arsobject_list:
+                    ars_dict['ars-object'] = {'arsobject': arsobject_list}
+
+            if ars_dict:
+                payload = {'openconfig-system-ext:adaptive-routing-switching': ars_dict}
+                request = {'path': ARS_PATH, 'method': PATCH, 'data': payload}
+
+        return request
+
+    def get_renamed_dict(self, cfg_dict):
+        """This method renames the keys in a dictionary by replacing underscores with hyphens"""
+        renamed_dict = {}
+        for key in cfg_dict:
+            new_key = key.replace('_', '-')
+            renamed_dict[new_key] = cfg_dict[key]
+
+        return renamed_dict
+
+    def get_delete_ars_requests(self, commands, delete_all):
+        """This method returns a list of delete requests generated from commands"""
+        requests = []
+
+        if not commands:
+            return requests
+        if delete_all:
+            requests.append({'path': ARS_PATH, 'method': DELETE})
+            return requests
+
+        if commands.get('ars_objects'):
+            for obj in commands['ars_objects']:
+                name = obj['name']
+                if len(obj) == 1:
+                    requests.append(self.get_delete_ars_object(name))
+                    continue
+                for key in obj:
+                    if key == 'name':
+                        continue
+                    if obj.get(key):
+                        attr = key.replace('_', '-')
+                        requests.append(self.get_delete_ars_object(name, attr))
+
+        if commands.get('port_bindings'):
+            for bind in commands['port_bindings']:
+                name = bind['name']
+                if len(bind) == 1:
+                    requests.append(self.get_delete_port_bind(name))
+                    continue
+                if bind.get('profile'):
+                    requests.append(self.get_delete_port_bind(name, 'profile'))
```

Comment:
For port-binding, both `interface` and `profile` are required. Hence if an interface name is specified (with or without the profile) delete that binding.
```suggestion
                requests.append(self.get_delete_port_bind(bind['name']))
```

Reply:
For port bindings, functionally 'interface' on its own has no usage without 'profile'. 

In CLI, interface config command requires the profile, while no config command deletes the binding for that interface directly. (profile is not needed for deletion)